src/util/d.ts

```diff
@@ -0,0 +1,43 @@
+import { ComposeFactory } from 'dojo-compose/compose';
+import {
+	DNode,
+	HNode,
+	WNode,
+	Widget,
+	WidgetState,
+	WidgetOptions
+} from 'dojo-interfaces/widgetBases';
+import { VNode, VNodeProperties } from 'dojo-interfaces/vdom';
+import { h } from 'maquette';
+
+export type TagNameOrFactory = string | ComposeFactory<Widget<WidgetState>, WidgetOptions<WidgetState>>;
+
+export type DOptions = VNodeProperties | WidgetOptions<WidgetState>;
+
+function d(tagName: string, options?: VNodeProperties, children?: (DNode | VNode)[]): HNode;
+function d(factory: ComposeFactory<Widget<WidgetState>, WidgetOptions<WidgetState>>, options: WidgetOptions<WidgetState>): WNode;
+function d(tagNameOrFactory: TagNameOrFactory, options: DOptions = {}, children?: (DNode | VNode)[]): DNode {
+	children = children ? children : [];
+
+	if (typeof tagNameOrFactory === 'string') {
+		children = children.filter((child) => child);
+
+		return {
+			children: children,
+			render(this: { children: VNode[] }) {
+				return h(<string> tagNameOrFactory, <VNodeProperties> options, this.children);
```

Comment:
We should note here that in 2.1 the cast of `tagNameOrFactory` will be unnecessary, because arguments that aren't touched within a function scope will be treated as `const` and therefore the narrowing above will cross the function barrier.